apis/python/src/tiledbsoma/_dense_nd_array.py

```diff
@@ -291,6 +291,9 @@ def write(
         """
         _util.check_type("values", values, (pa.Tensor,))
 
+        if np.isnan(values).any():
+            raise SOMAError("soma_data cannot contain nulls")
```

Comment:
Do we want to be able to write nans for floating point arrays? Not as a validity buffers, but as a regular data value.